routers/web/user/code.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package user
+
+import (
+	"net/http"
+
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	code_indexer "code.gitea.io/gitea/modules/indexer/code"
+	"code.gitea.io/gitea/modules/setting"
+)
+
+const (
+	tplUserCode base.TplName = "user/code"
+)
+
+// Code render user/organization code search page
+func Code(ctx *context.Context) {
+	if !setting.Indexer.RepoIndexerEnabled {
+		ctx.Redirect(ctx.ContextUser.HTMLURL())
+		return
+	}
+
+	ctx.Data["IsPackageEnabled"] = setting.Packages.Enabled
+	ctx.Data["IsRepoIndexerEnabled"] = setting.Indexer.RepoIndexerEnabled
+	ctx.Data["Title"] = ctx.Tr("code.title")
+	ctx.Data["ContextUser"] = ctx.ContextUser
+
+	language := ctx.FormTrim("l")
+	keyword := ctx.FormTrim("q")
+
+	queryType := ctx.FormTrim("t")
+	isMatch := queryType == "match"
+
+	if keyword != "" {
+		var (
+			repoIDs []int64
+			err     error
+		)
+
+		page := ctx.FormInt("page")
+		if page <= 0 {
+			page = 1
+		}
+
+		repoIDs, err = repo_model.FindUserCodeAccessibleUserRepoIDs(ctx.ContextUser.ID, ctx.Doer)
+		if err != nil {
+			ctx.ServerError("SearchResults", err)
+			return
+		}
+
+		var (
+			total                 int
+			searchResults         []*code_indexer.Result
+			searchResultLanguages []*code_indexer.SearchResultLanguages
+		)
+
+		if len(repoIDs) > 0 {
+			total, searchResults, searchResultLanguages, err = code_indexer.PerformSearch(ctx, repoIDs, language, keyword, page, setting.UI.RepoSearchPagingNum, isMatch)
+			if err != nil {
+				if code_indexer.IsAvailable() {
+					ctx.ServerError("SearchResults", err)
+					return
+				}
+				ctx.Data["CodeIndexerUnavailable"] = true
+			} else {
+				ctx.Data["CodeIndexerUnavailable"] = !code_indexer.IsAvailable()
+			}
```

Comment:
I am fairly certain that these lines can be rewritten into a single line.
However, I'm too tired at the moment to attempt that.

Reply:
not really this was discussed previously already in initial PR for code search